In [1]:
# importar bibliotécas de trabajo

import pandas as pd

In [2]:
# leer archivos de datos

goog = pd.read_csv("D:\\datos\\Entrenamiento\\IntitutoTecnologicoTelefonica\\ProgramaSuperiorBigData\\Modulo_10_Proyecto\\GOOG.csv",
                    sep = ",")

fb = pd.read_csv("D:\\datos\\Entrenamiento\\IntitutoTecnologicoTelefonica\\ProgramaSuperiorBigData\\Modulo_10_Proyecto\\FB.csv",
                    sep = ",")

amzn = pd.read_csv("D:\\datos\\Entrenamiento\\IntitutoTecnologicoTelefonica\\ProgramaSuperiorBigData\\Modulo_10_Proyecto\\AMZN.csv",
                    sep = ",")

msft = pd.read_csv("D:\\datos\\Entrenamiento\\IntitutoTecnologicoTelefonica\\ProgramaSuperiorBigData\\Modulo_10_Proyecto\\MSFT.csv",
                    sep = ",")

btc_usd = pd.read_csv("D:\\datos\\Entrenamiento\\IntitutoTecnologicoTelefonica\\ProgramaSuperiorBigData\\Modulo_10_Proyecto\\BTC_USD.csv",
                    sep = ",")

#dias = pd.read_excel("D:\\datos\\Entrenamiento\\IntitutoTecnologicoTelefonica\\ProgramaSuperiorBigData\\Modulo_10_Proyecto\\DIAS.xlsx")



In [3]:
goog.head()

Date        Open        High         Low       Close   Adj Close  \
0  2014-09-17  578.421936  585.911377  577.190308  583.168884  583.168884   
1  2014-09-18  585.392822  587.925842  583.398254  587.656616  587.656616   
2  2014-09-19  589.880493  594.846863  587.885986  594.447937  594.447937   
3  2014-09-22  592.194153  592.322754  581.862488  585.761780  585.761780   
4  2014-09-23  585.243225  585.243225  579.409241  579.538879  579.538879   

    Volume  
0  1692800  
1  1444500  
2  3736600  
3  1689500  
4  1471400

In [4]:
amzn.head()

Date        Open        High         Low       Close   Adj Close  \
0  2014-09-17  327.760010  328.000000  321.519989  324.000000  324.000000   
1  2014-09-18  325.440002  327.100006  323.709991  325.000000  325.000000   
2  2014-09-19  327.600006  332.760010  325.570007  331.320007  331.320007   
3  2014-09-22  328.489990  329.489990  321.059998  324.500000  324.500000   
4  2014-09-23  322.459991  327.600006  321.250000  323.630005  323.630005   

    Volume  
0  4136000  
1  2576900  
2  6886200  
3  3109700  
4  2352600

In [5]:
msft.head()

Date       Open       High        Low      Close  Adj Close  \
0  2014-09-17  46.259998  46.689999  46.230000  46.520000  41.066914   
1  2014-09-18  46.590000  46.830002  46.459999  46.680000  41.208172   
2  2014-09-19  46.810001  47.570000  46.599998  47.520000  41.949707   
3  2014-09-22  47.299999  47.380001  46.980000  47.060001  41.543625   
4  2014-09-23  46.849998  46.980000  46.470001  46.560001  41.102226   

      Volume  
0   38311900  
1   35556600  
2  202522400  
3   38686100  
4   33430300

In [6]:
fb.head()

Date       Open       High        Low      Close  Adj Close    Volume
0  2014-09-17  75.959999  77.250000  75.570000  76.430000  76.430000  28693000
1  2014-09-18  76.769997  77.330002  76.519997  77.000000  77.000000  23031000
2  2014-09-19  77.400002  78.300003  76.440002  77.910004  77.910004  76671000
3  2014-09-22  77.000000  77.529999  75.949997  76.800003  76.800003  31961000
4  2014-09-23  76.330002  78.629997  76.040001  78.290001  78.290001  36814000

In [7]:
btc_usd.head()

Date        Open        High         Low       Close   Adj Close  \
0  2014-09-17  465.864014  468.174011  452.421997  457.334015  457.334015   
1  2014-09-18  456.859985  456.859985  413.104004  424.440002  424.440002   
2  2014-09-19  424.102997  427.834991  384.532013  394.795990  394.795990   
3  2014-09-20  394.673004  423.295990  389.882996  408.903992  408.903992   
4  2014-09-21  408.084991  412.425995  393.181000  398.821014  398.821014   

     Volume  
0  21056800  
1  34483200  
2  37919700  
3  36863600  
4  26580100

In [8]:
# Selección de los datos de interés


goog_cierre = goog[["Date","Close"]]

fb_cierre = fb[["Date","Close"]]

amzn_cierre = amzn[["Date","Close"]]

msft_cierre = msft[["Date","Close"]]

btc_usd_cierre = btc_usd[["Date","Close"]]



In [10]:
# Renombrar columna de cierre de cada DataFrame, para que no se confundan al hacer el merge

goog_cierre = goog_cierre.rename(columns = {"Close" : "CloseGoog"})
fb_cierre = fb_cierre.rename(columns = {"Close" : "CloseFb"})
amzn_cierre = amzn_cierre.rename(columns = {"Close" : "CloseAmzn"})
msft_cierre = msft_cierre.rename(columns = {"Close" : "CloseMsft"})
btc_usd_cierre = btc_usd_cierre.rename(columns = {"Close" : "ClosBtcEur"})

# Establecer el índice en la columna Date para cada archivo

goog_cierre.set_index('Date', inplace = True)
fb_cierre.set_index('Date', inplace = True)
amzn_cierre.set_index('Date', inplace = True)
msft_cierre.set_index('Date', inplace = True)
btc_usd_cierre.set_index('Date', inplace = True)


In [11]:
goog_cierre.head()

CloseGoog
Date                  
2014-09-17  583.168884
2014-09-18  587.656616
2014-09-19  594.447937
2014-09-22  585.761780
2014-09-23  579.538879

In [12]:
fb_cierre.head()

CloseFb
Date                 
2014-09-17  76.430000
2014-09-18  77.000000
2014-09-19  77.910004
2014-09-22  76.800003
2014-09-23  78.290001

In [13]:
amzn_cierre.head()

CloseAmzn
Date                  
2014-09-17  324.000000
2014-09-18  325.000000
2014-09-19  331.320007
2014-09-22  324.500000
2014-09-23  323.630005

In [14]:
msft_cierre.head()

CloseMsft
Date                 
2014-09-17  46.520000
2014-09-18  46.680000
2014-09-19  47.520000
2014-09-22  47.060001
2014-09-23  46.560001

In [16]:
btc_usd_cierre.head()

ClosBtcEur
Date                  
2014-09-17  457.334015
2014-09-18  424.440002
2014-09-19  394.795990
2014-09-20  408.903992
2014-09-21  398.821014

In [18]:
# Integración de todos los dataframe en uno solo dataframe

df_combinado = btc_usd_cierre.join(goog_cierre).join(amzn_cierre).join(fb_cierre).join(msft_cierre)
df_combinado.head()


ClosBtcEur   CloseGoog   CloseAmzn    CloseFb  CloseMsft
Date                                                                
2014-09-17  457.334015  583.168884  324.000000  76.430000      46.52
2014-09-18  424.440002  587.656616  325.000000  77.000000      46.68
2014-09-19  394.795990  594.447937  331.320007  77.910004      47.52
2014-09-20  408.903992         NaN         NaN        NaN        NaN
2014-09-21  398.821014         NaN         NaN        NaN        NaN

In [20]:
df_combinado.describe().transpose()

count         mean          std         min         25%  \
ClosBtcEur  2304.0  5006.556875  5028.177584  178.102997  445.919991   
CloseGoog   1587.0   987.871607   316.886451  491.201416  733.770019   
CloseAmzn   1587.0  1329.917391   807.451260  286.950012  671.235016   
CloseFb     1587.0   154.719011    53.024936   72.629997  112.694999   
CloseMsft   1587.0    97.494058    52.015881   40.290001   53.809999   

                    50%          75%           max  
ClosBtcEur  3904.584961  8488.242920  34741.707031  
CloseGoog   1007.039978  1188.289978   1827.989990  
CloseAmzn   1125.349976  1819.224976   3531.449951  
CloseFb      155.270004   185.715003    303.910004  
CloseMsft     83.180000   131.795006    231.649994

# Comentario respecto al resumen descriptivo

Se observa como en los precios de cierre de las empresas el número de registros válidoses de 1587 casos esto es 329 casos perdidos respecto al precio de cierre del Bitcoin (2304), estos valores perdidos se corresponden a los días en que no hay actividad en las bolsas de valores, por lo que se sustituiran los mismos por el valor de cierre del último día hábil antes del valor perdido.

La sustitución de los valores perdidos se hace en el siguiente código

In [21]:
# Extracción de las filas y las columnas para usarlas como auxiliaraes para sustituir valores perdidos


filas = df_combinado.shape[0] 
columnas = df_combinado.shape[1]

df_combinado_null = df_combinado.isnull()

''' Con este par de loops se recorre el dataframe identificandose los valores perdidos y sustituyendolos por el valor de la fila
    anterior, que se corresponde que el valor de cierre para cada compañía el día anterior. 
    La lógica que sigue esta sustitución es la siguiente: Los valores perdidos se corresponden a días donde no hay actividad de 
    las casas de cambio, por lo que el valor asignado se corresponde''' 

for c in range(columnas):
    for f in range(filas):
        if df_combinado_null.iloc[f,c] == True:
            df_combinado.iloc[f,c] = df_combinado.iloc[(f-1),c]
        else:
            df_combinado.iloc[f,c] = df_combinado.iloc[f,c]

df_combinado.head()            

ClosBtcEur   CloseGoog   CloseAmzn    CloseFb  CloseMsft
Date                                                                
2014-09-17  457.334015  583.168884  324.000000  76.430000      46.52
2014-09-18  424.440002  587.656616  325.000000  77.000000      46.68
2014-09-19  394.795990  594.447937  331.320007  77.910004      47.52
2014-09-20  408.903992  594.447937  331.320007  77.910004      47.52
2014-09-21  398.821014  594.447937  331.320007  77.910004      47.52

In [22]:
df_combinado.describe().transpose()



count         mean          std         min         25%  \
ClosBtcEur  2304.0  5006.556875  5028.177584  178.102997  445.919991   
CloseGoog   2304.0   988.458379   317.691941  491.201416  733.775024   
CloseAmzn   2304.0  1329.480004   806.596225  286.950012  672.640015   
CloseFb     2304.0   154.714718    53.082751   72.629997  112.655002   
CloseMsft   2304.0    97.581918    52.067325   40.290001   53.762501   

                    50%          75%           max  
ClosBtcEur  3904.584961  8488.242920  34741.707031  
CloseGoog   1010.169983  1188.194977   1827.989990  
CloseAmzn   1126.500000  1818.597504   3531.449951  
CloseFb      155.070007   185.767505    303.910004  
CloseMsft     83.260002   131.642505    231.649994

El resultado anterio, muestra que los valores perdidos fueron reemplazados, respecto a los valores de los estadísticos descriptivos, los mismos no presentaron modificaciones significativas